rotkehlchen/api/rest.py

```diff
@@ -2730,21 +2710,55 @@ def delete_blockchain_transaction_data(
         return api_response(OK_RESULT, status_code=HTTPStatus.OK)
 
     @async_api_call()
-    def refresh_evm_transactions(
+    def refresh_transactions(
             self,
-            filter_query: EvmTransactionsFilterQuery,
+            from_timestamp: Timestamp,
+            to_timestamp: Timestamp,
+            accounts: list[BlockchainAccountData] | None,
     ) -> dict[str, Any]:
-        chain_ids: tuple[SUPPORTED_CHAIN_IDS]
-        if filter_query.chain_id is None:
-            chain_ids = get_args(SUPPORTED_CHAIN_IDS)
-        else:
-            chain_ids = (filter_query.chain_id,)
+        blockchain_addresses: dict[SupportedBlockchain, ListOfBlockchainAddresses]
+        with self.rotkehlchen.data.db.conn.read_ctx() as cursor:
+            if accounts is None or len(accounts) == 0:
+                blockchain_addresses = {
+                    chain: addr_list for chain in CHAINS_WITH_TRANSACTIONS
+                    if len(addr_list := self.rotkehlchen.data.db.get_single_blockchain_addresses(
+                        cursor=cursor,
+                        blockchain=chain,
+                    )) != 0
+                }
+            else:
+                blockchain_addresses = defaultdict(list)
+                for account in accounts:
+                    chains = (
+                        [account.chain] if account.chain is not None else
+                        self.rotkehlchen.data.db.get_account_blockchains(
+                            cursor=cursor,
+                            account=account.address,
+                        )
+                    )
+                    for chain in chains:
+                        blockchain_addresses[chain].append(account.address)  # type: ignore[arg-type]  # accounts with the same chain will have the same type of address
```

Comment:
why not try batching the accounts without explicit blockchains into a single query with an `IN` clause?
this way, you avoid n+1 queries when you have many accounts without specified blockchains.

something like...
```txt
explicit_accounts = accounts with blockchain specified
implicit_accounts = accounts without blockchain

for entry in explicit account:
    blockchain_addresses[chain].append(address).

if len(implicit accounts) > 0:
    query = "SELECT account, blockchain FROM blockchain_accounts WHERE account IN (?...)"
    ...